### ISSUE TITLE
LSTM and F1 Score metrics calculation problem in classification task 

### ISSUE THREAD

**Issue by JoesSattes** [cainmagi/MDNT#1]
I tried modify demo-classification.ipynb file to use TimeDistributed layer, Flatten layer and LSTM layer before Dense layer. It can training but accuracy is very decrease and can't calculate F1 Score metric with keras backend.

F1 score calculate by this code.
```python
def fscore(y_true, y_pred):
    if K.sum(K.round(K.clip(y_true, 0, 1))) == 0:
        return 0

    p = precision(y_true, y_pred)
    r = recall(y_true, y_pred)
    f_score = 2 * (p * r) / (p + r + K.epsilon())
    return f_score
```

Model architecture code.
```python    
    input_img = tf.keras.layers.Input(shape=input_shape)

    norm_input = Normalize(axis=-1)(input_img) #4feature axis=1, 1feature axis=-1
    conv_1 = mdnt.layers.AConv2D(channel_1, 8, padding='same', **cwkwargs)(norm_input)
    for i in range(blockLen):
        conv_1 = Block(channel_1, (3,3), **kwargs)(conv_1)
    conv_2 = Block(channel_2, 8, dropout=dropout, **kwargs)(conv_1)
    for i in range(blockLen):
        conv_2 = Block(channel_2, (3,3), **kwargs)(conv_2)
    conv_3 = Block(channel_3, 8, dropout=dropout, **kwargs)(conv_2)

    if kwargs['activation'] == 'prelu':
        conv_4 = tf.keras.layers.PReLU(shared_axes=[1,2])(conv_3)
    else:
        conv_4 = tf.keras.layers.Activation(kwargs['activation'])(conv_3)
    pool_1 = tf.keras.layers.MaxPooling2D((3,3))(conv_4)
    pool_1 = tf.keras.layers.Dropout(0.25)(pool_1)

    flat_1 = tf.keras.layers.Flatten()(pool_1)
    
    flat_1 = tf.keras.layers.TimeDistributed(tf.keras.layers.Flatten())(pool_1) #(pool_1)
    
    flat_1 = tf.keras.layers.LSTM(channel_4)(flat_1)
```

Model complie with fscore metrics
```python
model.compile(optimizer=optm, loss=tf.keras.losses.categorical_crossentropy, metrics=[tf.keras.metrics.categorical_accuracy, fscore])
```
